src/lang/mongolog/fluents.pl

```diff
@@ -62,6 +63,13 @@
 	assertz(mongolog_fluent(Functor, ArgFields, TimeField, Options)),
 	mongolog:add_command(Functor).
 
+
+mongolog_update_fluent_options(Functor, NewOptions) :-
+	mongolog_fluent(Functor, ArgFields, TimeField, Options),
+	retract(mongolog_fluent(Functor, ArgFields, TimeField, Options)),
```

Comment:
problem for concurrent processing here.
Prolog supports "Transactions": https://www.swi-prolog.org/pldoc/man?section=transactions